### ISSUE TITLE
Cell Execute Reply

### ISSUE THREAD

**Issue by emeeks** [nteract/archived-desktop-app#5162]
Is executionStateIdle and executeReply being attached to the cell via redux anywhere? I can see the timing of the idle/busy states in metadata and I can see the error response in outputs and that seems to be how you're determining an error but we should also be able to examine executeReply in case there's an error that doesn't give an output.

**Comment by captainsafia** [comment]
Yes, the execution reply that is sent from the kernel is stored in the Redux store via the execution epics ([ref](https://github.com/nteract/nteract/blob/master/packages/epics/src/execute/execute.ts#L141)).

Whenever we get a reply from the kernel, we check to see if the message type is associated with an output. These types are [defined here](https://github.com/nteract/nteract/blob/master/packages/messaging/src/index.ts#L208).

We take the same approach with kernel statuses, updating them from the execution epics ([ref](https://github.com/nteract/nteract/blob/master/packages/epics/src/execute/execute.ts#L125)).

The execution epics do currently have some error handling for common errors. You can see where the errors are handled by searching for `catchError` in [this file](https://github.com/nteract/nteract/blob/master/packages/epics/src/execute/execute.ts).

Let me know if you have any follow-up questions.

**Comment by MSeal** [comment]
If it help, I think the intention here was to actually capture the execution_reply message because the outputs are the source of truth for if there was an execution error. E.g. a kernel could return an error output but not fail execution. Additionally if there's ZMQ buffer overflow issues you may never get an error output (also some less mature kernels don't behave well with error outputting), but you'll likely get the execute_reply because they use different buffers.

**Comment by captainsafia** [comment]
@MSeal I see. The content above should largely hold true. Here's a more complete example of implementing an epic that extracts all `execute_reply` messages emitted by the kernel.

```js
const watchExecute =   (action$: Observable<
    actions.NewKernelAction | actions.KillKernelSuccessful
  >
) =>
  // Global message watcher so we need to set up a feed for each new kernel
  action$.pipe(
    ofType(actions.LAUNCH_KERNEL_SUCCESSFUL),
    switchMap((action) => {
        return action.payload.kernel.channels.pipe(
            ofMessageType("execute_reply"),
            map((msg) => console.log(msg)));
    });
```
